plugins/inputs/uwsgi/uwsgi.go

```diff
@@ -0,0 +1,231 @@
+// Package uwsgi implements a telegraf plugin for
+// collecting uwsgi stats from the uwsgi stats server.
+package uwsgi
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+// uWSGI Server struct
+type Uwsgi struct {
+	Servers []string `toml:"server"`
+	Timeout int      `toml:"timeout"`
+}
+
+// Errors is a list of errors accumulated during an interval.
+type Errors []error
+
+func (errs Errors) Error() string {
+	s := ""
+	for _, err := range errs {
+		if s == "" {
+			s = err.Error()
+		} else {
+			s = s + ". " + err.Error()
+		}
+	}
+	return s
+}
+
+// NestedError wraps an error returned from deeper in the code.
+type NestedError struct {
+	// Err is the error from where the NestedError was constructed.
+	Err error
+	// NestedError is the error that was passed back from the called function.
+	NestedErr error
+}
+
+// Error returns a concatenated string of all the nested errors.
+func (ne NestedError) Error() string {
+	return ne.Err.Error() + ": " + ne.NestedErr.Error()
+}
+
+// Errorf is a convenience function for constructing a NestedError.
+func Errorf(err error, msg string, format ...interface{}) error {
+	return NestedError{
+		NestedErr: err,
+		Err:       fmt.Errorf(msg, format...),
+	}
+}
+
+// Description returns the plugin description
+func (u *Uwsgi) Description() string {
+	return "Read uWSGI metrics."
+}
+
+// SampleConfig returns the sample configuration
+func (u *Uwsgi) SampleConfig() string {
+	return `
+    ## List with urls of uWSGI Stats servers. Url must match pattern:
+    ## scheme://address[:port]
+    ##
+    ## For example:
+    ## servers = ["tcp://localhost:5050", "http://localhost:1717", "unix:///tmp/statsock"]
+    servers = []
+    timeout = 5
+`
+}
+
+// Gather collect data from uWSGI Server
+func (u *Uwsgi) Gather(acc telegraf.Accumulator) error {
+	var errs Errors
+	for _, s := range u.Servers {
```

Comment:
This will need to be done concurrently across all servers.  Alternatively, you could have the plugin handle only a single server and then users can just configure multiple instances of this plugin.  This was is easier, and I think it is usually better unless there is a large amount of duplicated configuration.